controllers/amphoracontroller_controller.go

```diff
@@ -418,12 +430,25 @@ func (r *OctaviaAmphoraControllerReconciler) generateServiceConfigMaps(
 	if err != nil {
 		return err
 	}
-	templateParameters["ServiceUser"] = instance.Spec.ServiceUser
+	caPassSecret, _, err := secret.GetSecret(
+		ctx, helper, instance.Spec.CAKeyPassphraseSecret, instance.Namespace)
+	if err != nil {
+		return err
+	}
+	spec := instance.Spec
+	templateParameters["ServiceUser"] = spec.ServiceUser
 	templateParameters["KeystoneInternalURL"] = keystoneInternalURL
 	templateParameters["KeystonePublicURL"] = keystonePublicURL
-	templateParameters["ServiceRoleName"] = instance.Spec.Role
+	templateParameters["ServiceRoleName"] = spec.Role
 	templateParameters["LbMgmtNetworkId"] = templateVars.LbMgmtNetworkID
 	templateParameters["AmpFlavorId"] = templateVars.AmphoraDefaultFlavorID
+	serverCAPassphrase := caPassSecret.Data["server-ca-passphrase"]
+	if serverCAPassphrase != nil {
+		templateParameters["ServerCAKeyPassphrase"] = string(serverCAPassphrase)
```

Comment:
The other secrets are passed through an Env var to the init script of the containers https://github.com/openstack-k8s-operators/octavia-operator/blob/main/templates/octaviaamphoracontroller/bin/init.sh#L49-L51

I'm wondering if there's a good reason for doing it, and if we should follow the same pattern

Reply:
As discussed in chat the init script pattern is considered obsolete and this is the correct way to do it now.